apollo-server-fastify's context function now receives reply object.

HW13 opened the pull request:
### Problem
The context function in `apollo-server-fastify` is not receiving the `reply` object. This is due to the `request` and `reply` being spread into `graphqlServerOptions`(in [resolveGraphqlOptions](https://github.com/apollographql/apollo-server/blob/01519bdb471e2684cfccbf275d88a8435c2bf50a/packages/apollo-server-core/src/graphqlOptions.ts#L91) via [runHttpQuery](https://github.com/apollographql/apollo-server/blob/01519bdb471e2684cfccbf275d88a8435c2bf50a/packages/apollo-server-core/src/runHttpQuery.ts#L124)) - but `graphqlServerOptions` is expecting an object ([here](https://github.com/apollographql/apollo-server/blob/945dd297feb818b90d7bde3f77759061a2ce2135/packages/apollo-server-core/src/ApolloServer.ts#L752)).

#### Related Issues
- https://github.com/apollographql/apollo-server/issues/3156

### Solution
Add `applyContextArgs` function which receives the `reply` and `request` being spread in, and places them into an object which is then passed into `graphqlServerOptions`.
 
### Test plan
Confirm unit tests pass:
```
$ cd packages/apollo-server-fastify && jest
```


Closes https://github.com/apollographql/apollo-server/issues/3156

apollo-cla commented:
@HW13: Thank you for submitting a pull request! Before we can merge it, you'll need to sign the Apollo Contributor License Agreement here: https://contribute.apollographql.com/

HW13 commented:
Here's a quick test to confirm the existence of this bug -> https://github.com/autotelic/apollo-fastify-bug-test. Cheers.

jsumners commented:
@HW13 it looks like this will also fix an issue where Apollo is stripping Fastify request decorators. Is that true?

HW13 commented:
I was able to access request decorators in my tests, but I don't think these changes had an affect on that. This PR just adds a function that nests the request and reply in an object before passing it to `graphqlServerOptions`, so any issues with the request object itself should still exist.

jsumners commented:
Thanks.

jsumners commented:
Has this been closed because the issue has been addressed?

abernix commented:
Unintentional closing! Please stand-by: https://github.com/apollographql/apollo-server/issues/4304

abernix commented on packages/apollo-server-fastify/src/ApolloServer.ts:
```diff
@@ -170,11 +170,17 @@ export class ApolloServer extends ApolloServerBase {
           }
 
+          const graphQLServerOptions = this.graphQLServerOptions.bind(this);
+          const applyContextArgs = async (
+            request?: FastifyRequest<IncomingMessage>,
+            reply?: FastifyReply<OutgoingMessage>,
+          ) => graphQLServerOptions({ request, reply });
```
For symmetry reasons, I think we should make the Fastify integration have define (almost exactly this, except with an explicit return type of `Promise<GraphQLOptions>`) as a `createGraphQLServerOptions` class method [like the other integrations](https://github.com/apollographql/apollo-server/search?q=%22createGraphQLServerOptions%22&unscoped_q=%22createGraphQLServerOptions%22) in the (this) Apollo Server repository.  Then, calling `super.graphQLServerOptions({ request, reply })` within that function to satisfy the return type.

HW13 commented:
Hey @abernix, thanks for the review! The changes you've request have been implemented. When you have a chance, could you give this another look? Cheers.

HW13 commented:
Quick update -> I needed to make some minor tweaks (`createGraphqlServerOptions` -> `createGraphQLServerOptions`). I've also squashed my commits and rebased with `main` - hence the force pushing. Hope that's all good, cheers. 😎 👍

HW13 commented:
Awesome, thanks @abernix! 😎 👍

abernix commented:
📣 Published in `apollo-server-fastify@2.16.0`!  Thanks so much for putting this together originally!  I appreciated your contribution efforts and follow-through.

Sytten commented:
@HW13 @abernix This is a breaking change and should have been a major bump. Previously we have the `req` parameter that we relied on that was passed to the context function. It is not true that nothing was passed. Now that the version is out, it is a bit too late to revert and push a major, but this should be documented in the README of the package so that people are aware.
